app/models/course/leaderboard_settings.rb

```diff
@@ -35,13 +35,48 @@ def display_user_count
     @settings.display_user_count || 30
   end
 
-  # Returns the number of users to be displayed on the leaderboard
+  # Set the number of users to be displayed on the leaderboard
   #
   # @param [Integer] count The number of users to be displayed
   def display_user_count=(count)
     @settings.display_user_count = count
   end
 
+  # Returns whether group leaderboard is enabled (disabled by default).
+  #
+  # @return [Boolean] option Setting on whether group leaderboard is enabled.
+  def display_group_leaderboard
+    @settings.display_group_leaderboard = false if @settings.display_group_leaderboard.nil?
+    @settings.display_group_leaderboard
+  end
+
+  # Enable or disable the option to display group leaderboard
+  #
+  # @param [Boolean|Fixnum] option Setting on whether group leaderboard is enabled.
+  #   By default, simple_form provides '0' and '1' for boolean fields.
+  #   The assignment method will handle this conversion to Boolean.
+  def display_group_leaderboard=(option)
+    if option.is_a?(String)
+      option = option == '1' ? true : false
```

Comment:
@weiqingtoh https://github.com/rails/rails/blob/4-2-stable/activerecord/lib/active_record/type/boolean.rb#L13 AR also does the type cast, I think this is fine. 

to be clear may be:
```
status = option == '1'
@settings.display_group_leaderboard = status
``` ?

And is there a need to check if the option is a `String` ?